### ISSUE TITLE
Better control over logicalNodeId assignment

### ISSUE THREAD

**Issue by lukasherman** [noctarius/snowcast#22]
Hello @noctarius,
We are working on idea to run two separate Hazelcast clusters with the same configuration in two datacenters. Each cluster would process the data separately, but datastore is shared between those two clusters. In order to achieve conflict-free processing, it is critical to ensure that IDs generated do not overlap across the clusters. 
Looking at the Snowcast code, and easy solution for that is to ensure that `logicalNodeId` is different for each Hazelcast node, regardless of cluster membership. So a basic idea is to control somehow `logicalNodeId` range for each of the clusters individually, for example first cluster would get values in 0-64 range, and second cluster would get 65-127 range. 
All that magic happens in [LogicalNodeTable](https://github.com/noctarius/snowcast/blob/master/src/main/java/com/noctarius/snowcast/impl/LogicalNodeTable.java) code, so passing `logicalNodeIdBaseOffset` there would work, I assume. I hope that I am not missing something important here.
I would appreciate your feedback on this.

regards
Lukas

**Comment by noctarius** [comment]
Yeah you're right. You could add a range (base, count) definition to define a certain range inside the available node range. It is an interesting idea to make the ID generation even more scalable and independent. Apart from that it would be possible to add a parameter to segment the nodeIds and select a specific segment. In this case the number of available nodeIds would have to be divisible by the given number of segments.

**Comment by lukasherman** [comment]
I like the idea to specify segment count and segment id, this allows to set sensible defaults for simple use cases (1 segment, id 1). I will take a look on that.

**Comment by lukasherman** [comment]
Closing as no longer relevant.
